### PR TITLE
Implement basic usage of table <caption> element

### PR DESCRIPTION
**Pull request by thevangelist**
screen shot 2017-10-04 at 15 45 12
Add tableCaption prop, which can be a component / string

Add simple test and an example to Storybook

### CODE REVIEW

**Comment by thevangelist** [comment]
Resolves: https://github.com/react-bootstrap-table/react-bootstrap-table2/issues/77

File: packages/react-bootstrap-table2-example/examples/basic/table-with-caption.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  text: 'Product ID'
+}, {
+  dataField: 'name',
+  text: 'Product Name'
+}, {
+  dataField: 'price',
+  text: 'Product Price'
+}];
+
+<BootstrapTableful keyField='id' data={ products } columns={ columns } />
```
Comment (khaninD) [inline packages/react-bootstrap-table2-example/examples/basic/table-with-caption.js]: Hi @thevangelist , **remove it** and **add**:
`const CaptionElement = () => <h3 style={{ borderRadius: '0.25em', textAlign: 'center', color: 'purple', border: '1px solid purple', padding: '0.5em' }}>Component as Header</h3>;
`

`<BootstrapTableful keyField="id" data={ products } tableCaption="Plain text header" columns={ columns } />
`

`<BootstrapTableful keyField="id" data={ products } tableCaption={<CaptionElement />} columns={ columns } />
`

File: packages/react-bootstrap-table2/src/caption.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+/* eslint react/require-default-props: 0 */
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const Caption = (props) => {
+  if (!props.children) { return null; }
```
Comment (khaninD) [inline packages/react-bootstrap-table2/src/caption.js]: @thevangelist , 
`if (!props.children) return null;`

**Comment by thevangelist** [comment]
@khaninD should be fixed now, can you check again?

File: packages/react-bootstrap-table2/src/bootstrap-table.js [hunk 3]
```diff
@@ -131,6 +134,10 @@ BootstrapTable.propTypes = {
   bordered: PropTypes.bool,
   hover: PropTypes.bool,
   condensed: PropTypes.bool,
+  tableCaption: PropTypes.oneOfType([
```
Comment (AllenFang) [inline packages/react-bootstrap-table2/src/bootstrap-table.js]: I think this props name can be more simple like `caption`, I think it's enough and easy to understand for developer.

**Comment by AllenFang** [comment]
BTW, @khaninD thanks your contribution :)

**Pull request merged**